app/models/interactive_page.rb

```diff
@@ -118,24 +118,19 @@ def interactive_page_items
     page_items.select{ |pi| Embeddable::is_interactive?(pi.embeddable) }
   end
 
-  def section_embeddables(section_title)
-    section = sections.find { |s| s.title == section_title }
+  def section_embeddables(section_id)
+    section = sections.find { |s| s.id == section_id }
```

Comment:
How about just passing the section object itself?
Since we are getting rid of the title lookup, visible_embeddables knows the section object, this would prevent the extra `find` call here.